client.go

```diff
@@ -51,6 +51,23 @@ func MaxPacketChecked(size int) ClientOption {
 	}
 }
 
+// UseFstat sets whether to use Fstat or Stat when File.WriteTo is called (usually when copying files).
+// Some servers limit the amount of open files and calling Stat after opening the file will throw an error
+// From the server. Setting this flag will call Fstat instead of Stat which is suppose to be called on an
+// open file handle.
+//
+// From our experience it happens often with IBM Sterling SFTP servers that have "extractability" level
+// set to 1 which means only 1 file can be open at any given time.
+//
+// If the server you are working with still has an issue with both Stat and Fstat calls you can always
+// open a file and just loop through multiple Read calls until the file is completely read.
```

Comment:
We should probably recommend `io.Copy(ioutil.Discard, file)` as a work around, which is semantically more clear of purpose, rather than recommending a looped Read.

Reply:
The problem is that some SFTP servers, specifically those on mainframes, have the notion that once a file has been completely read it gets deleted, so doing `io.Copy(ioutil.Discard, file)` will get you the size, but the SFTP server will assume you have read it all and delete it :-|

Reply:
Makes sense and 🤦‍♀ of course it does that. Although, I’m not saying we should do that in order to find the file size, but rather replace the comment here suggesting “loop through multiple Read calls until the file is completely read.”